File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        st = set(nums)
        ans = 0

        for num in st:
            if (num - 1) not in st:
                length = 1
                while (num + length) in st:
                    length += 1
                    ans = max(ans, length)

        return ans

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const freq = new Map()
        for (const num of nums){ // O(N)
            freq.set(num,((freq.get(num) || 0)+1));
           
        }
        const minHeap = new MinPriorityQueue(element=>element[1]);
        for(let [key , value] of freq.entries()){ 
            minHeap.enqueue([key, value]);
            if(minHeap.size()>k)
            minHeap.dequeue();
          
        }
        let ans = [];
        for(let i = 0 ; i < k ; i++){
            const [key,freq] = minHeap.dequeue();
            ans.push(key);

        }
        return ans;
        




    }
}
